api/src/main/java/io/strimzi/api/kafka/model/authentication/KafkaClientCustomAuthentication.java

```diff
@@ -0,0 +1,54 @@
+package io.strimzi.api.kafka.model.authentication;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+public class KafkaClientCustomAuthentication extends KafkaClientAuthentication {
+
+    public static final String TYPE_CUSTOM = "custom";
+    private String saslMechanism;
+    private String securityProtocol;
+    private String saslJaasConfig;
+    private String saslLoginCallbackHandlerClass;
+
+    @Override
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getType() {
+        return TYPE_CUSTOM;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getSaslMechanism() {
+        return saslMechanism;
+    }
+
+    public void setSaslMechanism(String saslMechanism) {
+        this.saslMechanism = saslMechanism;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public String getSecurityProtocol() {
+        return securityProtocol;
+    }
+
+    public void setSecurityProtocol(String securityProtocol) {
+        this.securityProtocol = securityProtocol;
+    }
```

Comment:
As discussed on Slack => this cannot be configured here. This has to be constructed in the operator.